src/test/java/uk/gov/hmcts/reform/sandl/snlevents/services/UserTransactionServiceTests.java

```diff
@@ -0,0 +1,147 @@
+package uk.gov.hmcts.reform.sandl.snlevents.services;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.UserTransaction;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.UserTransactionData;
+import uk.gov.hmcts.reform.sandl.snlevents.model.usertransaction.UserTransactionRulesProcessingStatus;
+import uk.gov.hmcts.reform.sandl.snlevents.model.usertransaction.UserTransactionStatus;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.UserTransactionDataRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.UserTransactionRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.service.RevertChangesManager;
+import uk.gov.hmcts.reform.sandl.snlevents.service.UserTransactionService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.AdditionalAnswers.returnsFirstArg;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+public class UserTransactionServiceTests {
+
+    @TestConfiguration
+    static class UserTransactionServiceTestContextConfiguration {
+
+        @Bean
+        public UserTransactionService userTransactionService() {
+            return new UserTransactionService();
+        }
+    }
+
+    @Autowired
+    private UserTransactionService userTransactionService;
+
+    @MockBean
+    private UserTransactionRepository userTransactionRepository;
+
+    @MockBean
+    private UserTransactionDataRepository userTransactionDataRepository;
+
+    @MockBean
+    private RevertChangesManager revertChangesManager;
+
+    private final UUID someId = new UUID(1, 1);
+    UserTransaction ut;
+
+    @Before
+    public void setUp() {
+        this.ut = new UserTransaction(someId,
+            UserTransactionStatus.STARTED,
+            UserTransactionRulesProcessingStatus.IN_PROGRESS);
+        when(userTransactionRepository.save(any(UserTransaction.class))).thenAnswer(returnsFirstArg());
+    }
+
+    @Test
+    public void getUserTransactionById_should_call_findOne_on_userTransactionRepository() {
+        this.userTransactionService.getUserTransactionById(someId);
+        verify(this.userTransactionRepository, times(1)).findOne(any(UUID.class));
```

Comment:
should be done with assert. mock the repository and then use assert instead of verify